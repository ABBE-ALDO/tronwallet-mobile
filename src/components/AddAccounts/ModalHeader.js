import React from 'react'
import { func, string } from 'prop-types'

import { View, Text } from '../Utils'
import FontelloButton from '../FontelloButton'
import { Colors } from '../DesignSystem'

const ModalHeader = ({ onClose, title }) => (
  <View align='center' vertical marginTop={15} marginBottom={20}>
    <View flex={1} marginBottom={3}>
      <FontelloButton onPress={onClose} name='close' size={13} />
    </View>
    <View flex={2} align='center'>
      <Text color={Colors.greyBlue}>{title}</Text>
    </View>
    <View flex={1} />
  </View>
)

ModalHeader.propTypes = {
  onClose: func,
  title: string.isRequired
}

ModalHeader.defaultProps = {
  onClose: () => {}
}

export default ModalHeader
