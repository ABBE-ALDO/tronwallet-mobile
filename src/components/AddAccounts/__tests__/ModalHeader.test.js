import React from 'react'
import { shallow } from 'enzyme'

import ModalHeader from '../ModalHeader'

jest.mock('../../../utils/i18n')

describe('Modal header component', () => {
  const setup = propOverrides => {
    const props = Object.assign({
      title: 'Unit test'
    }, propOverrides)

    const wrapper = shallow(
      <ModalHeader {...props} />
    )

    return {
      props,
      wrapper
    }
  }

  test('Should match when render component', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })
})
