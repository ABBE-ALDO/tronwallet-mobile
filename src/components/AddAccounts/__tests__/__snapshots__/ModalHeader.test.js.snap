// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal header component Should match when render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalHeader
    onClose={[Function]}
    title="Unit test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "align": "center",
      "children": Array [
        <Styled(View)
          align="stretch"
          flex={1}
          justify="flex-start"
          marginBottom={3}
        >
          <FontelloButton
            color="#FFFFFF"
            name="close"
            onPress={[Function]}
            padding={0}
            size={13}
          />
        </Styled(View)>,
        <Styled(View)
          align="center"
          flex={2}
          justify="flex-start"
        >
          <Styled(Text)
            color="#9c9eb9"
            font="medium"
            size="small"
          >
            Unit test
          </Styled(Text)>
        </Styled(View)>,
        <Styled(View)
          align="stretch"
          flex={1}
          justify="flex-start"
        />,
      ],
      "justify": "flex-start",
      "marginBottom": 20,
      "marginTop": 15,
      "vertical": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "stretch",
          "children": <FontelloButton
            color="#FFFFFF"
            name="close"
            onPress={[Function]}
            padding={0}
            size={13}
          />,
          "flex": 1,
          "justify": "flex-start",
          "marginBottom": 3,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#FFFFFF",
            "name": "close",
            "onPress": [Function],
            "padding": 0,
            "size": 13,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "children": <Styled(Text)
            color="#9c9eb9"
            font="medium"
            size="small"
          >
            Unit test
          </Styled(Text)>,
          "flex": 2,
          "justify": "flex-start",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Unit test",
            "color": "#9c9eb9",
            "font": "medium",
            "size": "small",
          },
          "ref": null,
          "rendered": "Unit test",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "stretch",
          "flex": 1,
          "justify": "flex-start",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "children": Array [
          <Styled(View)
            align="stretch"
            flex={1}
            justify="flex-start"
            marginBottom={3}
          >
            <FontelloButton
              color="#FFFFFF"
              name="close"
              onPress={[Function]}
              padding={0}
              size={13}
            />
          </Styled(View)>,
          <Styled(View)
            align="center"
            flex={2}
            justify="flex-start"
          >
            <Styled(Text)
              color="#9c9eb9"
              font="medium"
              size="small"
            >
              Unit test
            </Styled(Text)>
          </Styled(View)>,
          <Styled(View)
            align="stretch"
            flex={1}
            justify="flex-start"
          />,
        ],
        "justify": "flex-start",
        "marginBottom": 20,
        "marginTop": 15,
        "vertical": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "stretch",
            "children": <FontelloButton
              color="#FFFFFF"
              name="close"
              onPress={[Function]}
              padding={0}
              size={13}
            />,
            "flex": 1,
            "justify": "flex-start",
            "marginBottom": 3,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#FFFFFF",
              "name": "close",
              "onPress": [Function],
              "padding": 0,
              "size": 13,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "children": <Styled(Text)
              color="#9c9eb9"
              font="medium"
              size="small"
            >
              Unit test
            </Styled(Text)>,
            "flex": 2,
            "justify": "flex-start",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Unit test",
              "color": "#9c9eb9",
              "font": "medium",
              "size": "small",
            },
            "ref": null,
            "rendered": "Unit test",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "stretch",
            "flex": 1,
            "justify": "flex-start",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
